pkg/config/autodiscover/autodiscover.go

```diff
@@ -142,11 +143,19 @@ func buildContainersFromPodResource(pr *PodResource) (containers []configsection
 		container.PodName = pr.Metadata.Name
 		container.ContainerName = containerResource.Name
 		container.NodeName = pr.Spec.NodeName
+		// This is to have access to the pod namespace
+		for _, cs := range pr.Status.ContainerStatuses {
+			if cs.Name == container.ContainerName {
+				container.ContainerUID = strings.TrimPrefix(cs.ContainerID, "cri-o://")
```

Comment:
a bit of comment here? i guess it's either cri-o or docker, not both at the same time? also, in my case, it's actually "containerd://" ...

Reply:
This is to strip the prefix in the containerid for both cri-o and docker. containerd is not tested/supported yet. I will add a comment to explain it.

Reply:
In what kind of setup do we expect "docker://" here? i thought docker-shim was deprecated.

Reply:
why not just split on '://'?

Reply:
docker:// is used for our current minikube github/local test

Reply:
ok for the split on ://

Reply:
> docker:// is used for our current minikube github/local test

ah, ok, I am just surprised that minikube still uses docker-shim

Reply:
so far this is the only one that I got to work with multus yet. I tried cri-o with minikube but it did not work

Reply:
Their doc says both cri-o and containerd are also supported: https://minikube.sigs.k8s.io/docs/handbook/config/#runtime-configuration
You probably don't have cri-o installed. containerd should work as it gets installed with docker.

Reply:
It looks like it is working but just switching to cri-o breaks something in the operator deployment(times out). Minikube takes care of cri-o install, this part looked ok. It might be a similar issue to the dns problem I found when trying virtualbox instead of docker with calico+multus. I could not figure out what is missing to make everything work with cri-o yet, we might just be missing an option in minikube.

Reply:
> ok for the split on ://

fixed